kedro/config/omegaconf_config.py

```diff
@@ -0,0 +1,262 @@
+        Dict[str, Any]:  A Python dictionary with the combined
+            configuration from all configuration files.
+    """
+
+    if not patterns:
+        raise ValueError(
+            "'patterns' must contain at least one glob "
+            "pattern to match config filenames against."
+        )
+
+    # In the first iteration of the OmegaConfLoader we'll keep the resolver turned-off.
+    # It's easier to introduce them step by step, but removing them would be a breaking change.
+    _clear_omegaconf_resolvers()
+
+    aggregate_config = []
+    processed_files = set()  # type: Set[Path]
+
+    for conf_path in conf_paths:
+        if not Path(conf_path).is_dir():
+            raise ValueError(
+                f"Given configuration path either does not exist "
+                f"or is not a valid directory: {conf_path}"
+            )
+
+        config_filepaths = _lookup_config_filepaths(
+            Path(conf_path), patterns, processed_files, _config_logger, True
+        )
+
+        new_conf = _load_configs_omegaconf(config_filepaths=config_filepaths)
+
+        aggregate_config.append(new_conf)
+        processed_files |= set(config_filepaths)
+
+    if not processed_files:
+        raise MissingConfigException(
+            f"No files of YAML or JSON format found in {conf_paths} matching the glob "
+            f"pattern(s): {patterns}"
+        )
+    return dict(OmegaConf.merge(*aggregate_config))
+
+
+def _load_configs_omegaconf(config_filepaths: List[Path]) -> Dict[str, Any]:
+    """Recursively load and merge all configuration files using OmegaConf, which satisfy
+    a given list of glob patterns from a specific path.
+
+    Args:
+        config_filepaths: Configuration files sorted in the order of precedence.
+
+    Raises:
+        ValueError: If 2 or more configuration files contain the same key(s).
+        BadConfigException: If configuration is poorly formatted and
+            cannot be loaded.
+
+    Returns:
+        Resulting configuration dictionary.
+
+    """
+
+    config = {}
+    aggregate_config = []
+    seen_file_to_keys = {}  # type: Dict[Path, AbstractSet[str]]
+
+    for config_filepath in config_filepaths:
+        try:
+            single_config = OmegaConf.load(config_filepath)
+            config = single_config
+        except (ParserError, ScannerError) as exc:
+            assert exc.problem_mark is not None
+            line = exc.problem_mark.line
+            cursor = exc.problem_mark.column
+            raise ParserError(
+                f"Invalid YAML file {config_filepath}, unable to read line {line}, "
+                f"position {cursor}."
+            ) from exc
+
+        _check_duplicate_keys(seen_file_to_keys, config_filepath, dict(single_config))
+        seen_file_to_keys[config_filepath] = single_config.keys()
+        aggregate_config.append(single_config)
+
+    if len(aggregate_config) > 1:
+        return dict(OmegaConf.merge(*aggregate_config))
+
+    return config
```

Comment:
This should be able to  handle case when len=1 ? Then the line 251 `return config` can be removed

Reply:
More importantly, is `OmegaConf.merge` a syntatic suger of merging a list of dictionary? Is it possible to remove `aggregated_config` altogether and just merging the dictionary at each iteration?

Reply:
`OmegaConf.merge` throws an assertion error if the list is empty: https://github.com/omry/omegaconf/blob/master/omegaconf/omegaconf.py#L267. As far as I understand it's not just syntactic sugar and does the merging of configs in a considerate way.

Reply:
`OmegaConf.merge` throws an assertion error if the list is empty: https://github.com/omry/omegaconf/blob/master/omegaconf/omegaconf.py#L267. As far as I understand it's not just syntactic sugar and does the merging of configs in a considerate way.